go/vt/vtgate/planbuilder/testdata/aggr_cases.json

```diff
@@ -539,6 +539,28 @@
       ]
     }
   },
+  {
+    "comment": "WITH ROLLUP on unsharded keyspaces",
+    "query": "select a, b, count(*) from unsharded group by a, b with rollup",
+    "plan": {
+      "QueryType": "SELECT",
+      "Original": "select a, b, count(*) from unsharded group by a, b with rollup",
+      "Instructions": {
+        "OperatorType": "Route",
+        "Variant": "Unsharded",
+        "Keyspace": {
+          "Name": "main",
+          "Sharded": false
```

Comment:
add a case where `with rollup` if pushed down to single shard on `sharded keyspace` works

Reply:
added both plan test and end-to-end test with this scenario

Reply:
this exposed an issue with the planner, so thank you for asking for it 🙏